src/Decode.fs

```diff
@@ -122,20 +108,10 @@ module Decode =
     /// Returns <c>Ok</c> if the decoder succeeds, otherwise <c>Error</c> with the error message.
     /// </returns>
     let fromString (decoder : Decoder<'T>) =
-        fun value ->
+        fun (value: string) ->
             try
-                let serializationSettings =
-                    new JsonSerializerSettings(
-                        DateParseHandling = DateParseHandling.None,
-                        CheckAdditionalContent = true
-                    )
-
-                let serializer = JsonSerializer.Create(serializationSettings)
-
-                use reader = new JsonTextReader(new StringReader(value))
-                let res = serializer.Deserialize<JToken>(reader)
-
-                fromValue "$" decoder res
+                let rootJsonElement = JsonDocument.Parse(value).RootElement
```

Comment:
This is the "entry point" where we parse the json blob into the abstract syntax tree using `System.Text.Json`